libbeat/template/template.go

```diff
@@ -97,6 +97,9 @@ func New(beatVersion string, beatName string, esVersion string, config TemplateC
 
 func (t *Template) load(fields common.Fields) (common.MapStr, error) {
 
+	dynamicTemplates = nil
```

Comment:
With this function being called each time that a client connects to Elasticsearch I would expect that this could lead to data races (especially if the number of works is greater than 1). So because these are globals I think a mutex is required for reads and writes.

Reply:
I added a lock directly above for the `load` method. All access to dynamicTemplates and defaultFields happens inside the processor and generator which both are triggered here. Generate could also be called from outside but in this case it's one of the Cobra commands so will not overlap with the running beat and only happening once.